### PR TITLE
add enabled state for annotation items

### PR DESCRIPTION
**Pull request by 3nids**
The idea is to be able to control visibility per items in an annotation layer.

### CODE REVIEW

**Comment by 3nids** [comment]
(this doesn't seem to work as expected, items are still displayed)

**Comment by nyalldawson** [comment]
@3nids is this planned to be API only? I'm unsure how we'd expose a way to re-enable items after they've been disabled in the gui...

**Comment by 3nids** [comment]
Yes, it's API only.

**Comment by 3nids** [comment]
Sorry, my bad, this is working as expected and ready to be merged.

**Comment by nyalldawson** [comment]
> Sorry, my bad, this is working as expected and ready to be merged.

image

**Comment by 3nids** [comment]
what would you expect as a test, a blank image check?

**Comment by nyalldawson** [comment]
@3nids 

Maybe rendering a layer containing a mix of enabled/disabled annotations

File: tests/src/python/test_qgsannotationlayer.py [hunk 1]
```diff
@@ -52,8 +54,16 @@
 class TestQgsAnnotationLayer(QgisTestCase):
 
     @classmethod
-    def control_path_prefix(cls):
-        return 'annotation_layer'
+    def setUpClass(cls):
```
Comment (nyalldawson) [inline tests/src/python/test_qgsannotationlayer.py:57]: Why revert all the modernization to this test?

**Comment by 3nids** [comment]
ok, now with a basic test

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
Approved -- but I think that mask image is out of date and should be removed? It seems to be no resemblance to the rendered image

**Pull request merged**